src/CaptureFile/CaptureFileTest.cpp

```diff
@@ -20,15 +20,127 @@ static constexpr const char* kNotAnAnswerString = "Some odd number, not the answ
+
+  auto output_stream_or_error = CaptureFileOutputStream::Create(temp_file_name);
+  ASSERT_TRUE(output_stream_or_error.has_value()) << output_stream_or_error.error().message();
+
+  std::unique_ptr<CaptureFileOutputStream> output_stream =
+      std::move(output_stream_or_error.value());
+
+  EXPECT_TRUE(output_stream->IsOpen());
+
+  orbit_grpc_protos::ClientCaptureEvent event1 =
+      CreateInternedStringCaptureEvent(kAnswerKey, kAnswerString);
+  orbit_grpc_protos::ClientCaptureEvent event2 =
+      CreateInternedStringCaptureEvent(kNotAnAnswerKey, kNotAnAnswerString);
+
+  auto write_result = output_stream->WriteCaptureEvent(event1);
+  ASSERT_FALSE(write_result.has_error()) << write_result.error().message();
+  write_result = output_stream->WriteCaptureEvent(event2);
+  ASSERT_FALSE(write_result.has_error()) << write_result.error().message();
+  auto close_result = output_stream->Close();
+  ASSERT_FALSE(close_result.has_error()) << close_result.error().message();
+
+  auto capture_file_or_error = CaptureFile::OpenForReadWrite(temporary_file.file_path());
+  ASSERT_TRUE(capture_file_or_error.has_value()) << capture_file_or_error.error().message();
+  std::unique_ptr<CaptureFile> capture_file = std::move(capture_file_or_error.value());
+
+  auto capture_section = capture_file->CreateCaptureSectionInputStream();
+
+  {
+    auto event_or_error = capture_section->ReadEvent();
+    ASSERT_TRUE(event_or_error.has_value()) << event_or_error.error().message();
+    ASSERT_EQ(event_or_error.value().event_case(), ClientCaptureEvent::kInternedString);
+    EXPECT_EQ(event_or_error.value().interned_string().key(), kAnswerKey);
+    EXPECT_EQ(event_or_error.value().interned_string().intern(), kAnswerString);
+  }
+
+  {
+    auto event_or_error = capture_section->ReadEvent();
+    ASSERT_TRUE(event_or_error.has_value()) << event_or_error.error().message();
+    ASSERT_EQ(event_or_error.value().event_case(), ClientCaptureEvent::kInternedString);
+    EXPECT_EQ(event_or_error.value().interned_string().key(), kNotAnAnswerKey);
+    EXPECT_EQ(event_or_error.value().interned_string().intern(), kNotAnAnswerString);
+  }
+}
+
+TEST(CaptureFile, CreateCaptureFileWriteAdditionalSectionAndReadMainSection) {
+  auto temporary_file_or_error = orbit_base::TemporaryFile::Create();
+  ASSERT_TRUE(temporary_file_or_error.has_value()) << temporary_file_or_error.error().message();
+  orbit_base::TemporaryFile temporary_file = std::move(temporary_file_or_error.value());
+
+  std::string temp_file_name = temporary_file.file_path().string();
+
+  auto output_stream_or_error = CaptureFileOutputStream::Create(temp_file_name);
+  ASSERT_TRUE(output_stream_or_error.has_value()) << output_stream_or_error.error().message();
+
+  std::unique_ptr<CaptureFileOutputStream> output_stream =
+      std::move(output_stream_or_error.value());
+
+  EXPECT_TRUE(output_stream->IsOpen());
+
+  orbit_grpc_protos::ClientCaptureEvent event1 =
```

Comment:
I guess you could extract the creation of the capture file into a helper method, as you already have the same code above.

Reply:
As discussed offline, lets keep it as is.

Reply:
Why does using `CHECK` hurt readability of failures? Because you can't use `<<`? Why not use `FAIL_IF`? We can add `FAIL_UNLESS` if you don't want to invert the condition. There's a bit of duplicated code here.

Reply:
yep, using FAIL_IF kind of solves it but it relies on the fact that our tests are isolated and if we use another environment which does not use isolation FAIL_IF aborts the executable, not just one test and that means other tests are not executed. It also feels awkward using FAIL_IF in place of ASSERT_*.

I think having a bit of duplication in tests is fine if doing otherwise results in worse readability of failure results.